File: devbricksx-java-annotations/src/main/java/com/dailystudio/devbricksx/annotations/Page.java
package com.dailystudio.devbricksx.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/*
 * Page is deprecated by Page in package com.dailystudio.devbricksx.annotations.data
 */
@Deprecated
@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.METHOD)
public @interface Page {

    final static int DEFAULT_PAGE_SIZE = 10;

    int pageSize() default DEFAULT_PAGE_SIZE;

}

File: devbricksx-annotations/src/main/java/com/dailystudio/devbricksx/annotations/data/Page.kt
package com.dailystudio.devbricksx.annotations.data

import com.dailystudio.devbricksx.annotations.InternalConstants

@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.FUNCTION)
annotation class Page(val pageSize: Int = DEFAULT_PAGE_SIZE) {

    companion object {
        const val DEFAULT_PAGE_SIZE = InternalConstants.DEFAULT_PAGE_SIZE
    }

}